ksql-engine/src/main/java/io/confluent/ksql/schema/inference/DefaultSchemaInjector.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.inference;
+
+import io.confluent.ksql.ddl.DdlConfig;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.SqlFormatter;
+import io.confluent.ksql.parser.tree.AbstractStreamCreateStatement;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.parser.tree.Literal;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.parser.tree.StringLiteral;
+import io.confluent.ksql.parser.tree.TableElement;
+import io.confluent.ksql.schema.inference.TopicSchemaSupplier.SchemaAndId;
+import io.confluent.ksql.schema.inference.TopicSchemaSupplier.SchemaResult;
+import io.confluent.ksql.serde.DataSource;
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.KsqlStatementException;
+import io.confluent.ksql.util.TypeUtil;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Schema;
+
+public class DefaultSchemaInjector implements SchemaInjector {
+
+  private final TopicSchemaSupplier schemaSupplier;
+
+  public DefaultSchemaInjector(final TopicSchemaSupplier schemaSupplier) {
+    this.schemaSupplier = Objects.requireNonNull(schemaSupplier, "schemaSupplier");
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T extends Statement> PreparedStatement<T> forStatement(
+      final PreparedStatement<T> statement
+  ) {
+    if (!(statement.getStatement() instanceof AbstractStreamCreateStatement)) {
```

Comment:
`null`s are evil and should be avoided in normal code.  So there's two rules I follow:

- null check constructor arguments, i.e. never let your objects get constructed in an invalid state.
- use the type system to enforce null checking i.e. return `Optional<Thing` rather than `Thing` if the return value could be null.

Hence, in this case,`PreparedStatement` checks its arguments and doesn't allow a null statement.